### ISSUE TITLE
[feature request] magic key/repeat key

### ISSUE THREAD

**Issue by throwaway-squared** [rvaiya/keyd#845]
hi, ive been using keyd for like 2 days now and ive got basically my whole qmk keyboard setup running all on my laptop keyboard which is pretty sick

that being said, from what i read, i do miss one feature from qmk and it would be cool if it could be implemented:
the magic key/repeat/alternate repeat key

basically it repeats whatever letter was last typed, so 'dd' would become 'd[repeat]'
you can also specify that after typing a letter, a different one shall be repeated, so 'ar' (which is hard for me to type on colemak) would become 'a[repeat]'

thank you :)

**Comment by nsbgn** [comment]
This sounds like a cool feature!

I do want to note that it is *somewhat* possible already with the `oneshotm` action. There will be a lot of duplication and drawbacks such as the loss of repeat-by-holding, which is why a dedicated feature would be nice, but still, if you're really missing it:

```
[main]
a = oneshotm(repeata, a)
b = oneshotm(repeatb, b)
# etcetera

[repeata]
. = a

[repeatb]
. = b

# etcetera
```

**Comment by throwaway-squared** [comment]
thank you for providing a way to do what i was looking for - will try this now, although as you said its kinda impractical to do for every key if you just want a normal repeat key

**Comment by oiiiiiiii** [comment]
The problem with the workaround is that keyd can only handle a certain amount of layers. If I add to many that way [this assertion](https://github.com/rvaiya/keyd/blob/19135668c20d3fa8c2a906d09e78c94003aae1cd/src/ini.c#L112) fails and keyd crashes

**Comment by oiiiiiiii** [comment]
But repeat would not allow to define a more general magic key, would it? Like `a *mag* -> ab` but `c *mag* -> cd`

Only `a *mag* -> aa` and `c *mag* -> cc` right? Or am I missing something?

**Comment by oiiiiiiii** [comment]
Yes, the problem is just that you quickly run out of layers with that method as I mentioned above. But the max number of layers can be increased quite easily when building from source

**Comment by oiiiiiiii** [comment]
Actually one more question: Is it intended that when I bind `a = oneshotm(repeata, a)` as suggested above only one 'a'  get printed when I hold the `a` key?
